tests/test_particle_sets.py

```diff
@@ -329,3 +330,26 @@ class MyParticle(ptype[mode]):
     assert np.all([p.v_once == 11.0 for p in pset])
     assert (V_once.shape == (npart, ))
     assert (V_once[0] == 1.)
+
+
+@pytest.mark.parametrize('mode', ['scipy', 'jit'])
+def test_variable_written_ondelete(fieldset, mode, tmpdir, npart=3):
+    filepath = tmpdir.join("pfile_on_delete_written_variables")
+
+    def move_west(particle, fieldset, time, dt):
+        tmp = fieldset.U[time, particle.lon, particle.lat, particle.depth]  # to trigger out-of-bounds error
+        particle.lon -= 0.1 + tmp
+
+    def DeleteP(particle, fieldset, time, dt):
+        particle.delete()
+
+    lon = np.linspace(0.05, 0.95, npart, dtype=np.float32)
+    lat = np.linspace(0.95, 0.05, npart, dtype=np.float32)
+    pset = ParticleSet(fieldset, pclass=ptype[mode], lon=lon, lat=lat)
+
+    outfile = pset.ParticleFile(name=filepath, write_ondelete=True, type="indexed")
+    pset.execute(move_west, runtime=1.1, dt=0.1, output_file=outfile,
+                 recovery={ErrorCode.ErrorOutOfBounds: DeleteP})
+    ncfile = Dataset(filepath+".nc", 'r', 'NETCDF4')
+    lon = ncfile.variables['lon'][:]
+    assert (lon.size == npart)
```

Comment:
In this test, all the particles are deleted.
Shouldn't we keep one in the domain, and check that this one is not written?